e2e/capl-cluster-flavors/kubeadm-full-capl-cluster/chainsaw-test.yaml

```diff
@@ -0,0 +1,249 @@
+              - name: URI
+                value: linode/instances
+              - name: FILTER
+                value: (to_string({"tags":($cluster)}))
+            content: |
+              set -e
+              curl -s \
+                -H "Authorization: Bearer $LINODE_TOKEN" \
+                -H "X-Filter: $FILTER" \
+                -H "Content-Type: application/json" \
+                "https://$TARGET_API/$TARGET_API_VERSION/$URI"
+            check:
+              ($error): ~
+              (json_parse($stdout)):
+                results: 2
+
+    # Test to check if Load Balancer and Cluster are up
+    - name: Check Load Balancer and Cluster
+      try:
+        - assert:
+            file: check-parent-resources.yaml
+      catch:
+        - describe:
+            cluster: ($cluster)
+            apiVersion: cluster.x-k8s.io/v1beta1
+            kind: Cluster
+            namespace: ($namespace)
+        - describe:
+            cluster: ($cluster)
+            apiVersion: infrastructure.cluster.x-k8s.io/v1alpha2
+            kind: LinodeCluster
+            namespace: ($namespace)
+
+    # Get the KUBECONFIG of the child cluster for later use
+    - name: Get child cluster kubeconfig
+      try:
+        - script:
+            env:
+              - name: CLUSTER
+                value: ($cluster)
+              - name: NAMESPACE
+                value: ($namespace)
+              - name: CLUSTERCTL_CONFIG
+                value: (env('CLUSTERCTL_CONFIG'))
+            content: |
+              set -e
+              clusterctl get kubeconfig $CLUSTER -n $NAMESPACE > kubeadm-full-cluster-kubeconfig.yaml
+            check:
+              ($error == null): true
+
+    # Test for child cluster resources
+    - clusters:
+        kubeadm-full-cluster:
+          kubeconfig: ./kubeadm-full-cluster-kubeconfig.yaml
+      name: Check child cluster resources
+      try:
+        - assert:
+            cluster: kubeadm-full-cluster
+            file: assert-child-cluster-deployments.yaml
+        - assert:
+            cluster: kubeadm-full-cluster
+            file: assert-child-cluster-daemonsets.yaml
+      catch:
+        - describe:
+            cluster: kubeadm-full-cluster
+            apiVersion: apps/v1
+            kind: Deployment
+            namespace: kube-system
+        - describe:
+            cluster: kubeadm-full-cluster
+            apiVersion: apps/v1
+            kind: DaemonSet
+            namespace: kube-system
+
+    # Test available StatefulSets
+    - clusters:
+        kubeadm-full-cluster:
+          kubeconfig: ./kubeadm-full-cluster-kubeconfig.yaml
+      name: Check the available StatefulSets
+      try:
+        - assert:
+            cluster: kubeadm-full-cluster
+            file: assert-child-cluster-statefulsets.yaml
+      catch:
+        - describe:
+            cluster: kubeadm-full-cluster
+            apiVersion: apps/v1
+            kind: StatefulSet
+            namespace: kube-system
```

Comment:
You can put deployments, daemonsets, and stateful set until same chainsaw step

Reply:
Not sure why you have deployments and daemonset together and statefulsets separate here

Reply:
I split it cause it felt easier to debug, when something goes wrong. If everything is in the same step, if something does go wrong, the logs are a bit too much to go through.

Reply:
Good idea! In that case, could you split up the deployments and daemonsets into their own chainsaw step?

Reply:
Oh yeah! I'll split them up as well